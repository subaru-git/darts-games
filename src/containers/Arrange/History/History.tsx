import React, { FC } from 'react';

type HistoryProps = {
  history: ArrangeResultModel[];
  user: User | null | undefined;
};

const History: FC<HistoryProps> = ({ history, user }) => {
  console.log(history);
  console.log(user);
  return <div>History</div>;
};

export default History;
