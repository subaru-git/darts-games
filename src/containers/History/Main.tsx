import React, { FC, useState } from 'react';
import { useLiveQuery } from 'dexie-react-hooks';
import Board from '@/containers/History/Board';
import { useAuth } from '@/contexts/AuthContext';
import { db } from '@/db/db';
import { readResult } from '@/lib/GameHistoryManager';
import MainTemplate from '@/templates/MainTemplate';

const Main: FC = () => {
  const [loading, setLoading] = useState(true);
  const user = useAuth();
  const gameHistory = useLiveQuery(async () => querier(setLoading)) || initialQuery;
  if (loading) return <MainTemplate label='history-main' isLoading />;
  return (
    <MainTemplate label='history-main'>
      <Board history={gameHistory} user={user} />
    </MainTemplate>
  );
};

const querier = async (setLoading: (isLoading: boolean) => void): Promise<GameResultModel> => {
  try {
    const cricketMarkUp = await readResult(db.cricketMarkUpResult);
    const eaglesEye = await readResult(db.eaglesEyeResult);
    const doubleTrouble = await readResult(db.doubleTroubleResult);
    const sweet16 = await readResult(db.sweet16Result);
    const topsAndTens = await readResult(db.topsAndTensResult);
    const twoDartCombinations = await readResult(db.twoDartCombinationsResult);
    const aroundTheCompass = await readResult(db.aroundTheCompassResult);
    const tonsUp = await readResult(db.tonsUpResult);
    const route64 = await readResult(db.route64Result);
    const eightyThrew = await readResult(db.eightyThrewResult);
    const shanghaiNights = await readResult(db.shanghaiNightsResult);
    const switchHitter = await readResult(db.switchHitterResult);
    const bullyBully = await readResult(db.bullyBullyResult);
    const treblesForShow = await readResult(db.treblesForShowResult);
    const arrange = await readResult(db.arrangeResult);
    const countUp = await readResult(db.countUpResult);
    setLoading(false);
    return {
      cricketMarkUp,
      eaglesEye,
      doubleTrouble,
      sweet16,
      topsAndTens,
      twoDartCombinations,
      aroundTheCompass,
      tonsUp,
      route64,
      eightyThrew,
      shanghaiNights,
      switchHitter,
      bullyBully,
      treblesForShow,
      arrange,
      countUp,
    };
  } catch (e) {
    return initialQuery;
  }
};

const initialQuery: GameResultModel = {
  cricketMarkUp: [],
  eaglesEye: [],
  doubleTrouble: [],
  sweet16: [],
  topsAndTens: [],
  twoDartCombinations: [],
  aroundTheCompass: [],
  tonsUp: [],
  route64: [],
  eightyThrew: [],
  shanghaiNights: [],
  switchHitter: [],
  bullyBully: [],
  treblesForShow: [],
  arrange: [],
  countUp: [],
};

export default Main;
