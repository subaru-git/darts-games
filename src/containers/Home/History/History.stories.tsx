/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import History from './History';

export default {
  title: 'Containers/Home/History',
  component: History,
} as ComponentMeta<typeof History>;

const Template: ComponentStory<typeof History> = (args) => <History {...args} />;

export const Default = Template.bind({});
Default.args = {};
