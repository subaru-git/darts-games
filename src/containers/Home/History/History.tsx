'use client';
import React, { FC } from 'react';
import Image from 'next/image';
import { useBreakpoint } from '@/hooks/tailwind';
import { useTranslations } from 'next-intl';

const History: FC = () => {
  const isMd = useBreakpoint('md');
  return <>{isMd ? <DesktopHistory /> : <MobileHistory />}</>;
};

const DesktopHistory: FC = () => {
  const t = useTranslations('Home.history');
  return (
    <div className='flex justify-center gap-8'>
      <Image
        src='/history.webp'
        alt='history image'
        style={{ objectFit: 'cover' }}
        width={640}
        height={360}
      />
      <div className='flex max-w-lg flex-col justify-center gap-4'>
        <h2 className='text-2xl font-bold'>{t('title')}</h2>
        <p className='whitespace-pre-wrap text-base'>{t('description')}</p>
        <p className='whitespace-pre-wrap text-base'>{t('disclaimer')}</p>
      </div>
    </div>
  );
};

const MobileHistory: FC = () => {
  const t = useTranslations('Home.history');
  return (
    <div className='flex flex-col justify-center gap-4'>
      <Image
        src='/history.webp'
        alt='history image'
        style={{ objectFit: 'cover' }}
        width={360}
        height={180}
      />
      <h2 className='text-center text-lg font-bold'>{t('title')}</h2>
      <p className='whitespace-pre-wrap text-sm'>{t('description')}</p>
      <p className='whitespace-pre-wrap text-sm'>{t('disclaimer')}</p>
    </div>
  );
};

export default History;
