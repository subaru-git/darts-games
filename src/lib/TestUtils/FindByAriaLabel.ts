export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
