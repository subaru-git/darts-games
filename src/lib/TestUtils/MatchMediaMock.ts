import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
