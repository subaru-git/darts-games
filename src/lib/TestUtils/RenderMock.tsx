import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
