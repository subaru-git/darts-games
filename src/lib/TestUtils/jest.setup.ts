import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
