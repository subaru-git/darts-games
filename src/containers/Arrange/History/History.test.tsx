import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { v4 as uuidv4 } from 'uuid';
import History from './History';
import { render } from '@/lib/TestUtils/RenderMock';

test('single out fat bull', async () => {
  const mockDate = new Date(2021, 0, 1, 1, 1, 1);
  jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
  const { container } = render(<History history={history} user={null} />);
  const user = userEvent.setup();
  const out = screen.queryAllByLabelText('single out')[0];
  await user.click(out);
  const select = screen.getByRole('combobox');
  const options = screen.getAllByRole('option');
  expect(select).toBeTruthy();
  expect(options.length).toBe(8);
  expect(container).toMatchSnapshot();
});

test('master out fat bull', async () => {
  const mockDate = new Date(2021, 0, 1, 1, 1, 1);
  jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
  const { container } = render(<History history={history} user={null} />);
  const user = userEvent.setup();
  const out = screen.queryAllByLabelText('master out')[0];
  jest.useRealTimers();
  await user.click(out);
  const select = screen.getByRole('combobox');
  const options = screen.getAllByRole('option');
  expect(select).toBeTruthy();
  expect(options.length).toBe(8);
  expect(container).toMatchSnapshot();
});

const history: ArrangeResultModel[] = [
  {
    // prettier-ignore
    result: [
      { target: 167, score: [['T20', 'T19', 'D-BULL']] },
      { target: 28, score: [['D14', '0', '0']] },
      { target: 161, score: [['T20', 'T17', 'D-BULL']] },
      { target: 86, score: [['D-BULL', 'D18', '0']] },
      { target: 153, score: [['T17', 'T17', 'T17']] },
      { target: 166, score: [['D-BULL', 'D-BULL', 'T20'], ['D3', '0', '0']]},
      { target: 82, score: [['D-BULL', 'D16', '0']] },
      { target: 63, score: [['D-BULL', '13', '0']] },
    ],
    uuid: uuidv4(),
    settings: {
      range: { x: 0, y: 0 },
      out: 'single',
      simulation: true,
      separate: false,
      hard: false,
      game: false,
      pro: false,
    },
    playedAt: new Date().toJSON(),
  },
  {
    // prettier-ignore
    result: [
      { target: 152, score: [['T20', 'T20', 'D16']] },
      { target: 25, score: [['9', 'D8', '0']] },
      { target: 142, score: [['S-BULL', 'S-BULL', 'T14']] },
      { target: 87, score: [['T19', 'D15', '0']] },
      { target: 154, score: [['S-BULL', 'S-BULL', 'T18']] },
      { target: 123, score: [['D-BULL', 'T19', 'D8']]},
      { target: 67, score: [['17', 'S-BULL', '0']] },
      { target: 14, score: [['D7', '0', '0']] },
    ],
    uuid: uuidv4(),
    settings: {
      range: { x: 0, y: 0 },
      out: 'master',
      simulation: true,
      separate: false,
      hard: false,
      game: false,
      pro: false,
    },
    playedAt: new Date().toJSON(),
  },
  {
    // prettier-ignore
    result: [
      { target: 150, score: [['T20', 'T20', 'D15']] },
      { target: 26, score: [['10', 'D8', '0']] },
      { target: 140, score: [['S-BULL', 'S-BULL', 'D20']] },
      { target: 82, score: [['S-BULL', 'D16', '0']] },
      { target: 10, score: [['D5', '0', '0']] },
      { target: 120, score: [['D-BULL', 'S-BULL', 'D10']]},
      { target: 68, score: [['D14', 'D20', '0']] },
      { target: 16, score: [['D8', '0', '0']] },
    ],
    uuid: uuidv4(),
    settings: {
      range: { x: 0, y: 0 },
      out: 'double',
      simulation: true,
      separate: false,
      hard: false,
      game: false,
      pro: false,
    },
    playedAt: new Date().toJSON(),
  },
];
