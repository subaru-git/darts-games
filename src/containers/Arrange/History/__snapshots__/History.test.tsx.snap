// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div>
    History
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
